doc/secure_boot/Secureboot_uefi_key_management.md

```diff
@@ -0,0 +1,217 @@
+   openssl x509 -inform der -in db-orig-2.der -out db-orig-2.pem
+   openssl x509 -inform der -in db-orig-3.der -out db-orig-3.pem
+4. Removal of db-orig-2.pem to be discarded
+5. Convert db-orig-1.pem and db-orig-3.pem to esl format
+   cert-to-efi-sig-list -g <GUID> db-orig-1.pem db-orig-1.esl
+   cert-to-efi-sig-list -g <GUID> db-orig-3.pem db-orig-3.esl
+6. Concatenate these two esl files 
+   cat db-orig-1.esl db-orig-3.esl > DB-new.esl
+7. Create authenticated variable and sign it
+   sign-efi-sig-list -g "<GUID>" -k KEK.key -c KEK.crt db DB-new.esl DB-new.auth
+8. Copy DB-new.auth to the system and use CLI to update UEFI database
+
+````
+
+### 4.4 Revoke keys
+Revoking a key from the UEFI database is done when the previously trusted key is no longer trustworthy. This action is taken when a key's security has been compromised.
+It is achieved by inserting the authenticated variable for the compromised key into the forbidden key list(DBx). 
+
+```
+Sample example:
+If db-orig-1.pem is compromised the generate an authenticated variable for db-orig-1.pem
+1. Convert ESL of pem cert.
+   cert-to-efi-sig-list -g <GUID> db-orig-3.pem db-orig-3.esl
+2. Create authenticated variable
+   sign-efi-sig-list -g "<GUID>" -k KEK.key -c KEK.crt dbx db-orig-3.esl DB-revoke.auth
+3. Copy DB-revoke.auth to the system and use CLI to update the UEFI database
+```
+
+### 4.5 Show keys
+This is to display the PK, KEK, DB, and DBx key lists.
+
+## 5 SONIC implementation
+The access of UEFI variables may vary from platform to platform depending on the specific implementation of UEFI firmware. Sonic will provide a common layer for accessing these variables and vendors can implement platform-specific parts. The SONIC will provide a plugin Python for vendors to implement these APIs and integrate them with the SONiC generic layer.
+
+### 5.1 Plugin class
+
+![image](https://github.com/sacnaik/SONiC/assets/25231205/a42cd039-0675-4a71-84e0-bce00eb2ca9d)
+
+### 5.2 OVMF based platform can use Linux implementation to access UEFI variable
+The Linux kernel provides a standard mechanism to interact with UEFI variables. It provides efivarfs filesystem that exposes the EFI runtime variables as a file. In order to manage these efi variables over sysfs requires mounting efivarfs filesystem. This enables users to access these UEFI variables like a file.
+
+
+<img src="https://github.com/sacnaik/SONiC/assets/25231205/1566c4d1-098f-4bde-b830-289e793e168d" width="400" hight="400">
+
+
+```
+sudo mount -t efivarfs none /sys/firmware/efi/efivars
+
+Ex: UEFI key variables represented as sysfs file
+
+/sys/firmware/efi/efivars/PK-<GUID>
+/sys/firmware/efi/efivars/KEK-<GUID>
+/sys/firmware/efi/efivars/DB-<GUID>
+/sys/firmware/efi/efivars/DBX-<GUID>
+```
+The platform plugin for this OVMF type UEFI can use either sysfs or efi-readvar and efi-updatevar to implement getVariable() and setVariable() methods.   
+   
+
+## 6 SONiC CLI commands
```

Comment:
What are the potential return values here?
Updatevar might fail (it being immutable, specific platform constraints, etc').
What is the expected design for these type of commands? As I see that they only receive a file and perform an action.